### PR TITLE
feat: add parallel determination of UBA fees

### PR DESCRIPTION
**Pull request by james-a-morris**

### CODE REVIEW

**Comment by pxrl** [comment]
@james-a-morris Any thoughts on making this public so that I can just compute the balancing fee directly?

https://github.com/across-protocol/sdk-v2/blob/e056dfd3c24df37e4bd2267bc93daf9da96b7643/src/clients/UBAClient/UBAClientAbstract.ts#L92

**Review by pxrl — APPROVED** [review]
LGTM!

File: src/clients/UBAClient/UBAClientAbstract.ts [hunk 1]
```diff
@@ -89,7 +90,7 @@ export abstract class BaseUBAClient {
    * @param refundChainId The chainId of the refund
    * @returns The balancing fee for the given token on the given chainId at the given block number
    */
-  protected async computeBalancingFee(
+  public async computeBalancingFee(
```
Comment (nicholaspai) [inline src/clients/UBAClient/UBAClientAbstract.ts:93]: It seems wasteful to pass in both deposit and refund chain id. Why not require user to pass in a single chain, and we assume that that is the deposit chain for a deposit and a refund for a refund?

**Review by nicholaspai — APPROVED** [review]
Just one suggestion!

**Pull request merged**